client/allocrunner/taskrunner/interfaces/lifecycle.go

```diff
@@ -16,4 +16,10 @@ type TaskLifecycle interface {
 
 	// Kill a task permanently.
 	Kill(ctx context.Context, event *structs.TaskEvent) error
+
+	// HasHandle returns true if the task runner has a handle to the task
+	// driver, which is useful for distinguishing restored tasks during
+	// prestart hooks. Note: prestart should be idempotent whenever possible
+	// to handle restored tasks safely; use this as an escape hatch.
+	HasHandle() bool
```

Comment:
Is this true only if the task is running?  If so, would `IsRunning()` be a more appropriate name? I also suspect we need to watch out for check-then-act races (e.g. process killed immediately after check) - I suspect in this context, it's fine.  Might be worth noting the issue in the comment.

Reply:
Thanks for pointing that out... it's probably worth me double-checking that if the driver handle goes away we can't NPE in the template runner. At that point we're in a non-working state for this task to be sure but it shouldn't blow up the client process.

I went back-and-forth on the name because we only know that there's a driver handle, not that the process is running and I didn't want future callers to be confused by that. But given the TOCTOU concern you've pointed out, I don't think the caller can assume all that much here except a best effort so I'll change the name and add some commentary about that.

Reply:
Name changed in 2da0b1f